src/tsf/TfEditses.cpp

```diff
@@ -243,34 +243,34 @@ HRESULT CEditSessionObject::_GetTextAndAttribute(TfEditCookie ec, ITfRange* rang
 
     if (FAILED(hr = _GetNoDisplayAttributeRange(ec, rangeIn,
                                                 guids, guid_size,
-                                                no_display_attribute_range))) {
+                                                no_display_attribute_range.get()))) {
         return hr;
     }
 
 
 
-    CComPtr<ITfReadOnlyProperty> propComp;
+    wil::com_ptr_nothrow<ITfReadOnlyProperty> propComp;
     if (FAILED(hr = pic->TrackProperties(guids, guid_size,       // system property
                                          NULL, 0,                // application property
                                          &propComp))) {
         return hr;
     }
 
 
-    CComPtr<IEnumTfRanges> enumComp;
+    wil::com_ptr_nothrow<IEnumTfRanges> enumComp;
     if (FAILED(hr = propComp->EnumRanges(ec, &enumComp, rangeIn))) {
         return hr;
     }
 
-    CComPtr<ITfRange>  range;
+    wil::com_ptr_nothrow<ITfRange>  range;
     while (enumComp->Next(1, &range, NULL) == S_OK) {
         VARIANT var;
         BOOL fCompExist = FALSE;
 
-        hr = propComp->GetValue(ec, range, &var);
+        hr = propComp->GetValue(ec, range.get(), &var);
         if (S_OK == hr) {
 
-            CComQIPtr<IEnumTfPropertyValue> EnumPropVal(var.punkVal);
+            wil::com_ptr_nothrow<IEnumTfPropertyValue> EnumPropVal(wil::try_com_query_nothrow<IEnumTfPropertyValue>(var.punkVal));

```

Comment:
You can use "try_query_to" style functions to avoid the redundant interface name.

Reply:
Changed to
```
wil::com_ptr_nothrow<IEnumTfPropertyValue> EnumPropVal;
if (wil::try_com_query_to(var.punkVal, &EnumPropVal)) {
```